distributed/config.py

```diff
@@ -24,8 +24,13 @@ def ensure_config_file(destination=dask_config_path):
                     os.mkdir(os.path.dirname(destination))
                 except FileExistsError:
                     pass
-            shutil.copy(default_path, destination)
-
+            # Atomically create destination
+            tmp = '%s.tmp.%d' % (destination, os.getpid())
+            shutil.copy(default_path, tmp)
+            try:
+                os.rename(tmp, destination)
+            except OSError:
+                os.unlink(tmp)
```

Comment:
Slight request to use `os.remove` instead. I was unfamiliar with `unlink`, and per the docs they are equivalent (https://docs.python.org/3.6/library/os.html#os.unlink).